src/platform_impl/apple/uikit/view.rs

```diff
@@ -525,18 +525,22 @@ impl WinitView {
 
             match phase {
                 UITouchPhase::Began => {
-                    ivars.fingers.set(ivars.fingers.get() + 1);
-                    let primary = match ivars.primary_finger.get() {
-                        Some(primary_id) => primary_id == touch_id,
-                        None => {
-                            debug_assert_eq!(
-                                ivars.fingers.get(),
-                                1,
-                                "number of fingers were not counted correctly"
-                            );
-                            ivars.primary_finger.set(Some(touch_id));
-                            true
-                        },
+                    let primary = if let UITouchType::Pencil = touch_type {
+                        true
+                    } else {
+                        ivars.fingers.set(ivars.fingers.get() + 1);
```

Comment:
Do we actually need all of this `set` logic? I think just checking the `finger_id == self.primary_finger` and only updating it, like it's on Android will work the same?

Reply:
If the primary finger is lifted while other fingers are pressed, it should still be possible to recognize the same finger again when its pressed. By that logic, we should only reset the primary finger ID when *all* fingers are lifted.

This isn't an issue on Android because `ACTION_DOWN` and `ACTION_UP` only get triggered on the primary finger. So which finger exactly is the primary finger, even if it was lifted during a multi-touch interaction, is already tracked for us by the OS.